src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyRegisterService.java

```diff
@@ -1,38 +1,30 @@
 package uk.gov.companieshouse.company_appointments.service;
 
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import uk.gov.companieshouse.api.metrics.RegistersApi;
-import uk.gov.companieshouse.company_appointments.api.CompanyMetricsApiService;
 import uk.gov.companieshouse.company_appointments.exception.BadRequestException;
-import uk.gov.companieshouse.company_appointments.exception.ServiceUnavailableException;
 
 @Service
 public class CompanyRegisterService {
-    private static final String PUBLIC_REGISTER = "public-register";
-
-    private final CompanyMetricsApiService companyMetricsApiService;
 
-    @Autowired
-    public CompanyRegisterService(CompanyMetricsApiService companyMetricsApiService) {
-        this.companyMetricsApiService = companyMetricsApiService;
-    }
+    private static final String PUBLIC_REGISTER = "public-register";
 
-    public boolean isRegisterHeldInCompaniesHouse (String registerType, String companyNumber) throws ServiceUnavailableException, BadRequestException {
-        RegistersApi registersApi = companyMetricsApiService.invokeGetMetricsApi(companyNumber).getData().getRegisters();
+    public boolean isRegisterHeldInCompaniesHouse(String registerType, RegistersApi registersApi)
+            throws BadRequestException {
```

Comment:
`BadRequestException` is a `RuntimeException` and so does not need to be declared in the `throws` clause

Reply:
Fixed in latest commit